src/render/View3D.cpp

```diff
@@ -498,6 +514,42 @@ void View3D::paintGL()
     glCheckError();
 }
 
+
+QImage View3D::renderScreenShot(int width, int height)
+{
+    if (m_badOpenGL)
+        return QImage();
+
+    double dPR = getDevicePixelRatio();
```

Comment:
That is, just keep what we have, but get `dPR` from a setting or a fixed supersampling (eg, dPR = 2).  Probably want to rename it as such.